csharp/src/Apache.Arrow/Arrays/StringArray.cs

```diff
@@ -91,5 +93,16 @@ public string GetString(int index, Encoding encoding = default)
                     return encoding.GetString(data, bytes.Length);
             }
         }
+
+        // IEnumerable methods
+        public new IEnumerator<string> GetEnumerator()
+        {
+            return Enumerable.Range(0, Length).Select(i => GetString(i)).GetEnumerator();
```

Comment:
Instead of using `Enumerable.Range`, can we instead build `Enumerator` structs for these? That will be better performing.

See https://github.com/dotnet/runtime/blob/388edb64f004e1c24b350edca43cf9fd2fdb7a4e/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L1145-L1215 for an example.

Reply:
Ok i was hesitating to do it